src/crypto/pkcs1.js

```diff
@@ -76,50 +75,46 @@ async function getPkcs1Padding(length) {
 /**
  * Create a EME-PKCS1-v1_5 padded message
  * @see {@link https://tools.ietf.org/html/rfc4880#section-13.1.1|RFC 4880 13.1.1}
- * @param {String} M message to be encoded
- * @param {Integer} k the length in octets of the key modulus
- * @returns {Promise<String>} EME-PKCS1 padded message
+ * @param {Uint8Array} message message to be encoded
+ * @param {Integer} keyLength the length in octets of the key modulus
+ * @returns {Promise<Uint8Array>} EME-PKCS1 padded message
  * @async
  */
-eme.encode = async function(M, k) {
-  const mLen = M.length;
+eme.encode = async function(message, keyLength) {
+  const mLength = message.length;
   // length checking
-  if (mLen > k - 11) {
+  if (mLength > keyLength - 11) {
     throw new Error('Message too long');
   }
   // Generate an octet string PS of length k - mLen - 3 consisting of
   // pseudo-randomly generated nonzero octets
-  const PS = await getPkcs1Padding(k - mLen - 3);
+  const PS = await getPkcs1Padding(keyLength - mLength - 3);
   // Concatenate PS, the message M, and other padding to form an
   // encoded message EM of length k octets as EM = 0x00 || 0x02 || PS || 0x00 || M.
-  return String.fromCharCode(0) +
-    String.fromCharCode(2) +
-    PS +
-    String.fromCharCode(0) +
-    M;
+  const encoded = new Uint8Array(keyLength);
+  // 0x00 byte
+  encoded[1] = 2;
+  encoded.set(PS, 2);
+  // 0x00 bytes
+  encoded.set(message, keyLength - mLength);
+  return encoded;
 };
 
 /**
  * Decode a EME-PKCS1-v1_5 padded message
  * @see {@link https://tools.ietf.org/html/rfc4880#section-13.1.2|RFC 4880 13.1.2}
- * @param {String} EM encoded message, an octet string
- * @returns {String} message, an octet string
+ * @param {Uint8Array} encoded encoded message bytes
+ * @returns {Uint8Array} message
  */
-eme.decode = function(EM) {
-  // leading zeros truncated by bn.js
-  if (EM.charCodeAt(0) !== 0) {
-    EM = String.fromCharCode(0) + EM;
-  }
-  const firstOct = EM.charCodeAt(0);
-  const secondOct = EM.charCodeAt(1);
+eme.decode = function(encoded) {
   let i = 2;
-  while (EM.charCodeAt(i) !== 0 && i < EM.length) {
+  while (encoded[i] !== 0 && i < encoded.length) {
     i++;
   }
   const psLen = i - 2;
-  const separator = EM.charCodeAt(i++);
-  if (firstOct === 0 && secondOct === 2 && psLen >= 8 && separator === 0) {
-    return EM.substr(i);
+  const separator = encoded[i++];
```

Comment:
Check that this does not through if `i = length`
Also add invalid padding tests